### PR TITLE
Token proof docs

### PR DESCRIPTION
**Pull request by MartinMinkov**

### CODE REVIEW

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 1]
```diff
@@ -94,17 +82,122 @@ In the snippet above, we define a smart contract called `SendExample` with a met
...
-### Understanding Important Terms
+## Authorization
+
+When a zkApp is interacting with a custom token that it did not originally create, the calling zkApp must get authorization from the [token owner](/zkapps/advanced-snarkyjs/custom-tokens#token-owner).
+
+There are two ways for a token owner to approve a transaction, one being a **signature** and the other being a **proof**.
+
+### Signature Authorization
+
+Signature authorization is the simplest way for a token owner to approve a custom token transfer however, it is the least flexible. If two seperate accounts want to trade a specific custom token, the token owner must sign the transaction before the transaction can be broadcasted.
```
[the remark this answers is not in the document]
Comment (jasongitmail) [reply docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:93]: `broadcasted` > `broadcast`

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 2]
```diff
@@ -94,17 +82,122 @@ In the snippet above, we define a smart contract called `SendExample` with a met
...
+
+Token zkApps should provide a `send` method, like the one described above. By providing a `send` method, the token zkApp can approve a transaction between two seperate accounts and sign it as the token owner.
+
+```ts
+tx = await Mina.transaction(feePayerKey, () => {
+  zkapp.sendTokens(account1Address, account2Address, UInt64.from(1_000)); // Sends 1,000 tokens from account1 to account2
```
[the remark this answers is not in the document]
Comment (jasongitmail) [reply docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:99]: Missing `let` or `const` in front of `tx`?

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 3]
```diff
@@ -94,17 +82,122 @@ In the snippet above, we define a smart contract called `SendExample` with a met
...
+  * This method is used to get authorization from the token owner. Remember,
+  * the token owner is the one who created the custom token, so to deduct our balance
+  * we must get authorization from the token owner.
+  */
+  @method approveSend(amount: UInt64) {
+    this.balance.subInPlace(amount);
+  }
+}
+
+tx = await Local.transaction(feePayer, () => {
```
Comment (jasongitmail) [inline docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:169]: missing `let` or `const`?

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 4]
```diff
@@ -121,8 +214,8 @@ Suppose a token account is being created for the first time. In that case, an ac
 
 Aside from sending custom tokens, custom tokens can be minted and burned by a **token owner account**. A token owner account is the governing zkApp account for a specific custom token.
 
-#### Token Owner
+### Token Owner
 
 A token owner is an account that creates, facilitates, and governs how a custom token is to be used. Concretely, the token owner is the account that created the custom token and is the only account that can mint and burn tokens.
 
-In addition to being the only account that can mint and burn tokens, the token owner is also the only account that can authorize sending tokens between two accounts. If two accounts want to send tokens to each other, the token owner must authorize the transaction. The token owner can authorize the transaction either with a signature or proof.
+In addition to being the only account that can mint and burn tokens, the token owner is also the only account that can approve sending tokens between two accounts. If two accounts want to send tokens to each other, the token owner must approve the transaction. The token owner can approve the transaction either with a signature or proof.
```
Comment (jasongitmail) [inline docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:221]: You're using the world `approve` in this paragraph, but `Signature Authorization` and `Proof Authorization` above. Intentional?

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 5]
```diff
@@ -94,17 +82,122 @@ In the snippet above, we define a smart contract called `SendExample` with a met
 
 For a more comprehensive example of how to use custom tokens with a zkApp, please see this [custom token example](https://github.com/o1-labs/snarkyjs/blob/main/src/lib/token.test.ts)
 
-### Understanding Important Terms
+## Authorization
+
+When a zkApp is interacting with a custom token that it did not originally create, the calling zkApp must get authorization from the [token owner](/zkapps/advanced-snarkyjs/custom-tokens#token-owner).
```
Comment (jasongitmail) [inline docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:87]: `is interacting` > `interacts`

**Review by jasongitmail — APPROVED** [review]
Very nice docs! Explains it really well. Left feedback for some minor items

**Comment by jasongitmail** [comment]
Note: Waiting to merge this until a newer version of SnarkyJS is published, which will contain features described in these docs.

File: docs/zkapps/advanced-snarkyjs/custom-tokens.mdx [hunk 6]
```diff
@@ -94,17 +82,122 @@ In the snippet above, we define a smart contract called `SendExample` with a met
...
+
+To allow for proof authorization by the token owner, the child zkApp that is requesting authorization must provide a way for the token owner to inspect the changes it wants to make and verify that they are valid. Token owner contracts have the power to inspect child account updates to enforce custom token rules. For example, a token owner contract could enforce that a child zkApp can only send tokens to a specific address.
+
+Token owner contracts can inspect the updates that a child zkApp wants to make by using a combination of `Experimental.Callback` and `this.experimental.approve`. The first thing that a token contract must do is generate the account updates that a child zkApp wants to make. The child zkApp wraps a function around `Experimental.Callback` which contains the changes it wants to make. The token owner can then execute that function with `this.approve` and inspect the changes that the child zkApp wants to make.
```
[the remark this answers is not in the document]
Comment (MartinMinkov) [reply docs/zkapps/advanced-snarkyjs/custom-tokens.mdx:113]: This is a good point! Made an issue [here](https://github.com/o1-labs/snarkyjs/issues/562) and tagged it for `To be Discussed`

**Pull request merged**